CheckBox: describe allowIndeterminateStateByClick (#4196)
DevExpress/devextreme-documentation: api-reference/10 UI Components/dxCheckBox/1 Configuration/allowIndeterminateStateByClick.md

---
---
##### shortDescription
Specifies whether users can set the CheckBox state to indeterminate.

---

> The component can be in three states (checked, unchecked, and indeterminate). When the CheckBox [value](/Documentation/ApiReference/UI_Components/dxCheckBox/Configuration/#value) is `null` or `undefined`, the default state is indeterminate.

Reviewer: "can be in"... The verb "to be" is considered a weak verb. I would replace it with something like "supports".

The you talk about the "default" state. I don't think it's 100% accurate. I guess I can set `value` to `null` in code for a number of reasons. So it's not only a possible "default" state, it's just a possible state. I understand that use this to introduce an example later, but I'll deal with that too. 

```suggestion
*dxCheckBox* supports three states: checked, unchecked, and indeterminate. The following table specifies the relation between the editor's state and its [value](/Documentation/ApiReference/UI_Components/dxCheckBox/Configuration/#value).

| Value | State |
| --- | --- |
| `null` or `undefined` | Indeterminate |
| `true` | Checked |
| `false` | Unchecked |

> If this property is set to `false`, users can change checkbox states in the following order:

Reviewer: I want to mention the property name. If you say "this property", then someone may think you're talking about  the `value` property that you mentioned in the previous paragraph. 

I think it's easier to start with the simplest example - all states are available. 
 
```suggestion
If you set *allowIndeterminateStateByClick* to `true`, users can cycles through check box states in the following order:
```

> *Indeterminate (default) → Checked → Unchecked → Checked → Unchecked.*

Reviewer: ```suggestion
*Indeterminate → Checked → Unchecked → Indeterminate →  ...*
```

> Set this property to `true` to allow users to set the indeterminate state:

Reviewer: Now we have arrived at a more complicated example. 

```suggestion
If you set *allowIndeterminateStateByClick* property to `false`, users can only cycle between the checked and unchecked states. Note that a check box can initially appear in the indeterminate state, if you haven't set the value to either `true` or `false`.
```

> *Indeterminate (default) → Checked → Unchecked → Indeterminate.*

Reviewer: ```suggestion
*Indeterminate → Checked → Unchecked → Checked → Unchecked → ...*
```
